python/ray/data/dataset.py

```diff
@@ -4255,6 +4256,61 @@ def write_datasink(
             datasink.on_write_failed(e)
             raise
 
+    def write_datasink_lazy(
+        self,
+        datasink: Datasink,
+        *,
+        prefilter_fn: Optional[Callable[[Block], Block]] = None,
```

Comment:
Could we remove the prefilter_fn for now to maintain consistency with write_datasink?